lib/filesystem/CArchiveLoader.h

```diff
@@ -94,6 +99,12 @@ class DLL_LINKAGE CArchiveLoader : public ISimpleResourceLoader
 
 	/** Holds all entries of the archive file. An entry can be accessed via the entry name. **/
 	std::unordered_map<ResourceID, ArchiveEntry> entries;
+
+	/** Specifies if Original H3 archives should be extracted to a separate folder **/
+	bool extractArchives;
 };
 
+/** Constructs the file path for the extracted file. Creates the subfolder hierarchy aswell **/
+boost::filesystem::path createExtractedFilePath(std::string outputSubFolder, std::string entryName);
+
```

Comment:
Declaration is excessive

Reply:
Can you elaborate please? I do not get it.